cli/src/torii/utils.rs

```diff
@@ -37,47 +35,25 @@ macro_rules! add_state {
 }
 
 pub mod body {
-    use iroha_core::smartcontracts::query::Error as QueryError;
-    use iroha_data_model::query::VersionedSignedQueryRequest;
+    use iroha_version::error::Error as VersionError;
 
     use super::*;
 
-    #[derive(Debug)]
-    #[allow(unused_tuple_struct_fields)]
-    pub struct WarpQueryError(QueryError);
-
-    impl From<QueryError> for WarpQueryError {
-        fn from(source: QueryError) -> Self {
-            Self(source)
-        }
-    }
-
-    impl warp::reject::Reject for WarpQueryError {}
-
-    impl TryFrom<&Bytes> for super::VerifiedQueryRequest {
-        type Error = WarpQueryError;
-
-        fn try_from(body: &Bytes) -> Result<Self, Self::Error> {
-            let res = VersionedSignedQueryRequest::decode_all_versioned(body.as_ref());
-            let query = res.map_err(|e| WarpQueryError(Box::new(e).into()))?;
-            let VersionedSignedQueryRequest::V1(query) = query;
-            Ok(Self::try_from(query)?)
-        }
-    }
-
-    /// Decode query request
-    pub fn query() -> impl Filter<Extract = (VerifiedQueryRequest,), Error = Rejection> + Copy {
-        warp::body::bytes()
-            .and_then(|body: Bytes| async move { (&body).try_into().map_err(warp::reject::custom) })
-    }
-
     /// Decode body as versioned scale codec
     pub fn versioned<T: DecodeVersioned>() -> impl Filter<Extract = (T,), Error = Rejection> + Copy
     {
         warp::body::bytes().and_then(|body: Bytes| async move {
             T::decode_all_versioned(body.as_ref()).map_err(warp::reject::custom)
         })
     }
+
+    /// Recover from failure in `versioned`
+    pub fn recover_versioned(rejection: Rejection) -> Result<impl Reply, Rejection> {
```

Comment:
Oh, no, everything is alright with out checking. It's in fact should be named `recover_signature()`, because `handle_queries()` can fail in `let request: VerifiedQueryRequest = request.try_into()?;` step, which is a signature check, not version

Reply:
I name this function like that because it is recovering **only** version errors which can pop up after call to `body::versioned`.

And errors inside `handle_queries` are directly returned to the user, because we have `impl Reply for WarpResult`.

Warp is confusing :)

Reply:
So this is an another minor fix? The main fix was done when you changed the paramter type of `handle_queries()`?

Reply:
Main fix was to move signature verification inside `handle_queries`.